assembl/static2/js/app/pages/debate.jsx

```diff
@@ -1,53 +1,123 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { isDateExpired } from '../utils/globalFunctions';
-import Survey from '../components/debate/survey';
-import Thread from '../components/debate/thread';
-import TwoColumns from '../components/debate/twoColumns';
-import TokenVote from '../components/debate/tokenVote';
+import { graphql } from 'react-apollo';
+import gql from 'graphql-tag';
+import { Translate } from 'react-redux-i18n';
+import Loader from '../components/common/loader';
+import Themes from '../components/debate/common/themes';
+import Timeline from '../components/debate/navigation/timeline';
+import Thumbnails from '../components/debate/navigation/thumbnails';
 
 class Debate extends React.Component {
-  getCurrentStepIdentifier() {
-    const currentDate = new Date();
-    const { debateData } = this.props.debate;
-    let identifier = null;
-    if (debateData.timeline){
-      debateData.timeline.map((phase) => {
-        const startDate = new Date(phase.start);
-        const endDate = new Date(phase.end);
-        if (isDateExpired(currentDate, startDate) && isDateExpired(endDate, currentDate)) {
-          identifier = phase.identifier;
-        }
-        return identifier;
-      });
-    }
-    return identifier || 'Thread';
+  constructor(props) {
+    super(props);
+    this.state = {
+      isThumbnailsHidden: true
+    };
+    this.showThumbnails = this.showThumbnails.bind(this);
+    this.hideThumbnails = this.hideThumbnails.bind(this);
+    this.displayThumbnails = this.displayThumbnails.bind(this);
+  }
+  showThumbnails() {
+    this.setState({ isThumbnailsHidden: false });
+  }
+  hideThumbnails() {
+    setTimeout(() => {
+      this.setState({ isThumbnailsHidden: true });
+    }, 400);
+  }
+  displayThumbnails() {
+    this.setState({ isThumbnailsHidden: !this.state.isThumbnailsHidden });
   }
   render() {
-    const currentIdentifier = this.getCurrentStepIdentifier();
+    const { loading, thematics } = this.props.data;
+    const { identifier, isNavbarHidden } = this.props;
+    const isParentRoute = !this.props.params.phase || false;
+    const themeId = this.props.params.themeId || null;
+    const children = React.Children.map(this.props.children, (child) => {
+      return React.cloneElement(child, {
```

Comment:
Yes, very important. The identifier is the name of a phase in the timeline.
I clone the child because this view is generic for each phase. The only thing that changes is the identifier. Look at the screenshot to understand.
<img width="368" alt="capture d ecran 2017-04-20 a 17 07 02" src="https://cloud.githubusercontent.com/assets/17748791/25237760/e06e7eb0-25eb-11e7-9ef7-948d518cbbc3.png">